### ISSUE TITLE
mAP zero only on yolov5m

### ISSUE THREAD

**Issue by dvivarelli** [ultralytics/yolov5#5902]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,
I'm trying to train yolov5m using 2000 images over 11 classes.
I'm using Torch 1.10 with CUDA 11.3.
I have already setup the dataset folder manually with data.yml, images and labels folder forntrsining and test.

When i train the medium model (yolov5m) i have mAP and labels equals to zero during training. Instead, if i use small model (yolov5s) everything seems to be fine (labels and mAP are not zero and mAP increase epoch by epoch)

Someone has an idea why? Which Is the difference (in terms of training dataset) between them?

**Comment by Zengyf-CVer** [comment]
@dvivarelli 
First, you should check the loss after training. If loss is not 0, the training is normal. Then expand the epochs and continue to observe.

**Comment by Zengyf-CVer** [comment]
@dvivarelli 
See:
https://github.com/ultralytics/yolov5/issues/5862#issuecomment-986548329

**Comment by dvivarelli** [comment]
This morning I tried again.
I suppose this is a bug when using pytorch CUDA 11.3 on Windows (home).

Using the same dataset, I have tried the following case:  
- training with pytorch 1.10 cuda 11.3  model yolov5m  --> mAP is always at 0 after 300 epochs
- training with pytorch 1.10 cuda 11.3  model yolov5s  --> mAP is always almost 0 after 100 epochs
- training using pytorch 1.10 CPU model yolov5m --> mAP is greater than 0 since the first step and increase over time
- training using pytorch 1.10 cuda 10.2 and yolov5m --> mAP is greater than 0 since the first step and increase over time

So i suppose there is something wrong with pytorch 1.10 and cuda 11.3

**Comment by Zengyf-CVer** [comment]
@dvivarelli 
This kind of bug does exist, you can try to install cuda11.1, I did an experiment on this, it is ok.

**Comment by dvivarelli** [comment]
just an update:
i tried with pytorch 1.9.0 and cuda 11.1 but the problem persist. 
I'm using a NVIDIA GTX 1650 TI and it supports cuda 11.5.

with cuda 10.2 evrything works fine.

**Comment by glenn-jocher** [comment]
@dvivarelli can you please be very detailed here: Are you saying on the same hardware and same CUDA11 environment all models work correctly except for YOLOv5m?

**Comment by glenn-jocher** [comment]
> * training with pytorch 1.10 cuda 11.3  model yolov5m  --> mAP is always at 0 after 300 epochs
> * training with pytorch 1.10 cuda 11.3  model yolov5s  --> mAP is always almost 0 after 100 epochs
> * training using pytorch 1.10 CPU model yolov5m --> mAP is greater than 0 since the first step and increase over time
> * training using pytorch 1.10 cuda 10.2 and yolov5m --> mAP is greater than 0 since the first step and increase over time

@dvivarelli or from these results it seems that no model works correctly on your hardware with CUDA11 right? So there's nothing special in this case about YOLOv5m and YOLOv5s, all other models will also fail the first two CUDA 11 trainings?

Other users have experienced similar CUDA11 issues with windows/conda and GTX GPUs. Typically downgrading to CUDA 10 resolves this issue. 

If you have any idea for a fix please let us know.

**Comment by pss3tssp** [comment]
i am getting 0 accuracy for nuclei segmentation even after 60 epochs ..i am using pytorch 1.11

**Comment by glenn-jocher** [comment]
@pss3tssp hello! Thank you for using YOLOv5 for nuclei segmentation.

To get an idea of what could be causing the issue, can you please provide more information? 

- Do you mean zero mean average precision (mAP) for your validation dataset?
- Is it the case that your model output itself is total black, which means, nothing is learned yet or other issues?
- What's the number of classes that you have and what's the number of images that you have in your dataset?

Also, if you could provide your `train.yaml` configuration file, some sample images and their labels, and a training log, it can really help to identify the cause of the problem.

Finally, please note that if your dataset contains only a few examples, it might be that you need to use a pre-trained model for transfer learning before starting your training. Please check our YOLOv5 [Tutorials](https://docs.ultralytics.com) for more information on this. 

Looking forward to hearing from you!
